nemo/core/classes/modelPT.py

```diff
@@ -676,6 +688,40 @@ def setup_optimizer_param_groups(self):
 
         self._optimizer_param_groups = param_groups
 
+    def setup(self, stage: Optional[str] = None):
```

Comment:
is there a better way to do it? What if user needs to overwrite setup for their model

Reply:
Other alternatives are more problematic. As far as I know, super() is a catch all empty method declaration in base PTL. If subclasses need to call ModelPT (this) impl, they can do so with super().

NLPModel overrides it currently without calling super, and that's fine, since they have their own way of deferring data setup. They may choose to call it (and ignore the processing of defer_setup arg) or not call it (and also ignore the defer_setup config argument).